### PR TITLE
 Make crate for axum

### PR DESCRIPTION
**Pull request by Dispersia**
Hello, I plan to use this crate with Axum, and just did all of the same logic (in axum's way) from the rocket/actix-web examples. If you prefer to not take on the burden of maintaining something that's so new and has constant new releases, I can close this PR and host it myself, just thought it might be useful for others too. Feel free to give any descriptions you would like for better error messaging :)
 
 Thanks

### CODE REVIEW

File: shaku_axum/tests/module_interface.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+use shaku::{module, Component, HasComponent, HasProvider, Interface, Provider};
```
Comment (AzureMarker) [inline shaku_axum/tests/module_interface.rs:1]: This should copy over the comment from the similar test in shaku_actix (modified for axum as required).

**Review by AzureMarker — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: shaku_axum/src/inject_provided.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+///
+/// # Example
+/// ```rust
+/// use axum::{routing::get, AddExtensionLayer, Router};
+/// use shaku::{module, Interface, Provider};
+/// use shaku_axum::InjectProvided;
+/// use std::net::SocketAddr;
+/// use std::sync::Arc;
+///
+/// trait HelloWorld: Interface {
```
[the remark this answers is not in the document]
Comment (Dispersia) [reply shaku_axum/src/inject_provided.rs]: Was my mistake, didn't realize provider didn't need it, but component did. However you were right, it did still need Send + Sync, so marked it accordingly. Thanks!

[the remark this answers is not in the document]
Comment (Dispersia) [reply shaku_axum/src/inject_provided.rs]: Just curious though on this comment, you stated only when "thread_safe" feature is enabled, but I had enabled it by default on shaku (just following the actix/rocket example), is there a case a user can disable that?

File: shaku_axum/src/inject_provided.rs [hunk 3]
```diff
@@ -72,22 +71,22 @@ where
...
-    type Rejection = (StatusCode, axum::Json<Value>);
+    type Rejection = (StatusCode, String);
 
     async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
         let module = get_module_from_state::<M, B>(req)?;
         let service = module.provide().map_err(|e| {
             (
                 axum::http::StatusCode::INTERNAL_SERVER_ERROR,
-                axum::Json(json!({ "error": format!("{}", e) })),
+                format!("{}", e),
```
Comment (AzureMarker) [inline shaku_axum/src/inject_provided.rs:81]: nit: this can use `to_string`:
```suggestion
                e.to_string(),
```

**Review by AzureMarker — APPROVED** [review]

**Comment by AzureMarker** [comment]
Thanks for the PR! I'll release the crate shortly.

**Comment by AzureMarker** [comment]
@Dispersia can I list you in the authors array as `Aaron Housh <dispersias@gmail.com>`?

**Comment by Dispersia** [comment]
Sure! You didn't have to do that, but thank you haha. This crate has been fantastic at helping my dotnet/scala friends move over to Rust, so thank you for it as well while I'm here :)

**Comment by AzureMarker** [comment]
shaku_axum 0.1.0 is released: https://crates.io/crates/shaku_axum/0.1.0

**Pull request merged**